.github/workflows/publish.yml

```diff
@@ -154,6 +154,27 @@ jobs:
         working-directory: arrow-libs
         run: ./gradlew publishTvosSimulatorArm64PublicationToMavenRepository
 
+  publish_watchos:
```

Comment:
Why did you split these? We can optimise this later, but we want to find a sweet spot where all jobs take aabout the same time, and we also want to cut back as much as possible on `macos-latest` because those are limited in how much can run in parallel. Whilst `ubuntu` we can run a lot of them for free in parallel.

So by optimising this (not needed in this PR) we can make more PRs, or branch run their action in parallel reducing wait time.

Reply:
it got cancled because it took to long

Reply:
Will reduce the duration once we have a example time frame